lib/Server.js

```diff
@@ -464,24 +468,23 @@ class Server {
           : // eslint-disable-next-line no-undefined
             undefined;
 
+      if (watchOptions.poll) {
+        delete watchOptions.poll;
+      }
```

Comment:
Why not get only valid options from `watchOptions`?

Reply:
Because we respect `poll` option for normalizing `usePolling` and `interval` option.

Reply:
Using this will modify webpack compiler options, you should not touch them

Reply:
You need `Default Options` + `Options from watchOptions` + `DeveloperOptions`, so developer can override even `usePolling` and `interval` (but by default we get them from `watchOptions`) and other

Reply:
Here we are not modifying `compiler.options.watchOptions`, only `devServer.static.watch` and `devServer.watchFiles.options`

Reply:
got it.

Reply:
Previously we were also respecting `devServer.static.watch.poll` which is not valid.

Reply:
We should respect them too, i.e.:
```
{
  ...defaultOptions,
  ...watchOptions
  ...developerOptionsFroChokidar
}
```

Reply:
This will pass `poll` to `chokidar` when developer defines `devServer.static.watch.poll`.